pyproject.toml

```diff
@@ -24,7 +24,7 @@ requires-python = ">=3.11"
 dependencies = [
     "sortedcontainers>=2.4.0",
     "toml>=0.10.2",
-    "z3-solver==4.12.6.0",
+    "z3-solver==4.13.4.0",
```

Comment:
can't remember why we pinned to 4.12 but can undo this if you want

Reply:
let's handle this separately, if possible.